src/js-ipfs/index.test.js

```diff
@@ -16,3 +16,28 @@ it('Should connect to js-ipfs', async (done) => {
   expect(opts.initJsIpfs.mock.calls.length).toBe(1)
   done()
 })
+
+it('Should connect to js-ipfs with custom config', async (done) => {
+  const mockIpfs = {}
+  const opts = {
+    ipfsConnectionTest: jest.fn().mockResolvedValueOnce(true),
+    getJsIpfs: () => ({
+      JsIpfs: jest.fn().mockResolvedValueOnce(true),
+      JsIpfsConfig: {
+        config: {
+          Addresses: {
+            Swarm: ['/dns4/ws-star.discovery.libp2p.io/tcp/443/wss/p2p-websocket-star']
+          }
+        }
+      }
+    }),
+    jsIpfsOpts: {},
+    initJsIpfs: jest.fn().mockResolvedValue(mockIpfs)
+  }
+  const res = await tryJsIpfs(opts)
+  expect(res.ipfs).toEqual(mockIpfs)
+  expect(res.provider).toEqual('js-ipfs')
+  expect(opts.ipfsConnectionTest.mock.calls.length).toBe(1)
+  expect(opts.initJsIpfs.mock.calls.length).toBe(1)
```

Comment:
We should test that the custom config gets passed to the jsipfs constructor; that the new thing that this feature adds. If you keep a ref to the jsipfs constructor mock, I believe you can assert that it was called with specific arguments.

Reply:
Hmm makes sense, but that is inside the `promiseMeJsIpfs` function that is outside the scope of the test. Not sure how to do that 🤔